src/transport/ResponseFuture.cpp

```diff
@@ -164,6 +171,39 @@ bool ResponseFuture::isTimeOut() const {
   return m_bAsync.load() == 1 && diff > m_timeout;
 }
 
+int ResponseFuture::getMaxRetrySendTimes() const {
+	return m_maxRetrySendTimes;
+} 
+int ResponseFuture::getRetrySendTimes() const {
+	return m_retrySendTimes;
+}
+
+void ResponseFuture::setMaxRetrySendTimes(int maxRetryTimes) {
+	m_maxRetrySendTimes = maxRetryTimes;
+}
+void ResponseFuture::setRetrySendTimes(int retryTimes) {
+	m_retrySendTimes = retryTimes;
+}
+
+void ResponseFuture::setBrokerAddr(const std::string& brokerAddr) {
+	m_brokerAddr = brokerAddr;
+}
+void ResponseFuture::setRequestCommand(const RemotingCommand& requestCommand) {
+	m_requestCommand = requestCommand;
+}
+
+const RemotingCommand& ResponseFuture::getRequestCommand() {
+	return m_requestCommand;
+}
+std::string ResponseFuture::getBrokerAddr() const {
+	return m_brokerAddr;
+}
+
+int64 ResponseFuture::leftTime() const {
```

Comment:
name this to `GetTimeoutBudget`?

Reply:
it doesn't matter

Reply:
But this name might make us confused actually right?